lib/suggestion-list.coffee

```diff
@@ -0,0 +1,112 @@
+{Model} = require 'theorist'
+{Emitter, CompositeDisposable} = require 'event-kit'
+
+module.exports =
+class SuggestionList extends Model
+  constructor: ->
+    @emitter = new Emitter
+    @subscriptions = new CompositeDisposable
+    # Allow keyboard navigation of the suggestion list
+    @subscriptions.add(atom.commands.add 'autocomplete-suggestion-list',
+      'autocomplete-plus:confirm': @confirmSelection,
+      'autocomplete-plus:select-next': @selectNext,
+      'autocomplete-plus:select-previous': @selectPrevious,
+      'autocomplete-plus:cancel': @cancel
+    )
+
+  addKeyboardInteraction: ->
+    @removeKeyboardInteraction()
+    keys =
+      'escape': 'autocomplete-plus:cancel'
+
+    completionKey = atom.config.get('autocomplete-plus.confirmCompletion') || ''
+    navigationKey = atom.config.get('autocomplete-plus.navigateCompletions') || ''
+
+
+    keys['tab'] = 'autocomplete-plus:confirm' if completionKey.indexOf('tab') > -1
+    keys['enter'] = 'autocomplete-plus:confirm' if completionKey.indexOf('enter') > -1
+
+    if @items?.length > 1 and navigationKey == 'up,down'
+      keys['up'] =  'autocomplete-plus:select-previous'
+      keys['down'] = 'autocomplete-plus:select-next'
+    else
+      keys['ctrl-n'] = 'autocomplete-plus:select-next'
+      keys['ctrl-p'] = 'autocomplete-plus:select-previous'
+
+    @keymaps = atom.keymaps.add(
+      'autocomplete-suggestion-list',
+      'atom-text-editor:not(.mini) .autocomplete-plus': keys
+    )
+
+    @subscriptions.add(@keymaps)
+
+  removeKeyboardInteraction: ->
+    @keymaps?.dispose()
+    @subscriptions.remove(@keymaps)
+
+  confirmSelection: =>
+    @emitter.emit('did-confirm-selection')
+
+  onDidConfirmSelection: (fn) ->
+    @emitter.on('did-confirm-selection', fn)
+
+  confirm: (match) =>
+    @emitter.emit('did-confirm', match)
+
+  onDidConfirm: (fn) ->
+    @emitter.on('did-confirm', fn)
+
+  selectNext: =>
+    @emitter.emit('did-select-next')
+
+  onDidSelectNext: (fn) ->
+    @emitter.on('did-select-next', fn)
+
+  selectPrevious: =>
+    @emitter.emit('did-select-previous')
+
+  onDidSelectPrevious: (fn) ->
+    @emitter.on('did-select-previous', fn)
+
+  cancel: =>
+    @emitter.emit('did-cancel')
+
+  onDidCancel: (fn) ->
+    @emitter.on('did-cancel', fn)
+
+  show: (editor) =>
+    return if @active
+    return unless editor?
+    @destroyOverlay()
+
+    marker = editor.getLastCursor()?.getMarker()
+    return unless marker?
+    @overlayDecoration = editor.decorateMarker(marker, { type: 'overlay', item: this })
+    @addKeyboardInteraction()
```

Comment:
It's definitely a bit weird, but it solves a defect where (for a single suggestion, where up/down arrows are really not helpful) a user wants to retain their ability to use the up / down arrows to move the cursor.